parsers/UY.py

```diff
@@ -1,163 +1,264 @@
-    salto_grande = get_salto_grande(session)
-    obj["Hidráulica"] = obj.get("Hidráulica", 0.0) + salto_grande
+    # helper func to format data output
+    def make_datum(entry):
+        return {
+            "zoneKey": zone_key,
+            "datetime": entry["time"].datetime,
+            "consumption": entry["demand"],
+            "source": "ute.com.uy",
+        }
 
-    return obj
+    # then delegate all the actual work to get_entry_list
+    return get_entry_list(session, make_datum, correct_hydro=False)
 
 
 def fetch_production(
     zone_key: str = "UY",
-    session: Optional[Session] = None,
+    session: Session = Session(),
     target_datetime: Optional[datetime] = None,
     logger: Logger = getLogger(__name__),
-) -> dict:
-    if target_datetime:
-        raise NotImplementedError("This parser is not yet able to parse past dates")
+) -> List[dict]:
 
-    obj = parse_page(session)
+    if target_datetime is not None:
+        raise ParserException(
+            parser="UY", message="This parser is unnable to parse dates in the past"
+        )
 
-    data = {
-        "zoneKey": zone_key,
-        "datetime": obj["datetime"],
-        "production": dict(
-            [(k, obj[INV_MAP_GENERATION[k]]) for k in INV_MAP_GENERATION.keys()]
-        ),
-        "source": "ute.com.uy",
-    }
+    # make a helper function to create output format
+    def make_datum(entry):
+        return {
+            "zoneKey": zone_key,
+            "datetime": entry["time"].datetime,
+            "production": {key: entry[key] for key in AVALIABLE_KEYS},
+            "source": "ute.com.uy",
+        }
 
-    return data
+    # then delegate all the actual work to get_entry_list
+    return get_entry_list(session, make_datum, correct_hydro=True)
 
 
 def fetch_exchange(
     zone_key1: str = "UY",
     zone_key2: str = "BR-S",
-    session: Optional[Session] = None,
+    session: Session = Session(),
     target_datetime: Optional[datetime] = None,
     logger: Logger = getLogger(__name__),
-) -> dict:
+) -> List[dict]:
     """Requests the last known power exchange (in MW) between two countries."""
 
-    session = session or Session()
-    if target_datetime:
-        raise NotImplementedError("This parser is not yet able to parse past dates")
+    if target_datetime is not None:
+        raise ParserException(
+            parser="UY", message="This parser is unnable to parse dates in the past"
+        )
 
     # set comparison
     if {zone_key1, zone_key2} != {"UY", "BR"}:
-        return None
+        raise ParserException(
+            parser="UY", message="This parser is unnable to parse dates in the past"
+        )
 
-    obj = parse_page(session)
-    netFlow = obj["Interconexión con Brasil"]  # this represents BR->UY (imports)
-    if zone_key1 != "BR":
-        netFlow *= -1
+    # flip directions if needed
+    direction_coeff = -1 if zone_key1 == "UY" else 1
```

Comment:
This is not needed since the parser only support `BR->UY` exchanges and the keys are always sorted alphabetically.

Reply:
I think this whole function in the parser is never used, cause the BR->UY transfer is handled on the brazil side/ Should I just delete this method for clarity?

Reply:
I'd keep it, might be good to have a backup if something happens with the Brazil parser.